pkg/cloudmap/enqueue_requests_for_pod_events.go

```diff
@@ -0,0 +1,80 @@
+package cloudmap
+
+import (
+	"context"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/k8s"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+func NewEnqueueRequestsForPodEvents(k8sClient client.Client, log logr.Logger) *enqueueRequestsForPodEvents {
+	return &enqueueRequestsForPodEvents{
+		k8sClient: k8sClient,
+		log:       log,
+	}
+}
+
+var _ handler.EventHandler = (*enqueueRequestsForPodEvents)(nil)
+
+type enqueueRequestsForPodEvents struct {
+	k8sClient client.Client
+	log       logr.Logger
+}
+
+// Create is called in response to an create event
+func (h *enqueueRequestsForPodEvents) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
+	h.enqueueVirtualNodesForPods(context.Background(), queue, e.Object.(*corev1.Pod))
+}
+
+// Update is called in response to an update event
+func (h *enqueueRequestsForPodEvents) Update(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
```

Comment:
ideally we should mirror endpointController's behavior here. (can be in follow up PR).
https://github.com/kubernetes/kubernetes/blob/3491129d348eb6fb5e4274112df5a946ded740af/pkg/controller/util/endpoint/controller_utils.go#L187
Things missing here:
1. we should check whether [pod's deletionTimestamp differs](https://github.com/kubernetes/kubernetes/blob/3491129d348eb6fb5e4274112df5a946ded740af/pkg/controller/util/endpoint/controller_utils.go#L168) we used deletionTimestamp in endpointResolver
2. we should check whether pod's labels changed, if pod labels changed.  if pod ready also changed, enqueue the union of vn(oldPod), vn(newPod). if pod ready didn't change, enqueue vn(oldPod), vn(newPod) (if they differs)